src/components/report-viewer-tab.js

```diff
@@ -66,43 +69,71 @@ export const ReportViewerTab = ({
         );
     }, [treeModel]);
 
+    const rootNodeId = useMemo(() => {
+        const rootNode = treeModel.treeNodes.find(
+            (node) => node?.data?.label === 'Root'
+        );
+        return rootNode?.id;
+    }, [treeModel]);
+
     const setNodeName = useCallback(
         (report) => {
-            report.defaultName =
-                report.taskKey === 'Root'
-                    ? report.taskKey
-                    : nodesNames.get(report.taskKey);
+            if (report.taskKey === 'Root') {
+                report.taskKey = rootNodeId;
+            } else {
+                report.defaultName = nodesNames.get(report.taskKey);
+            }
             return report;
         },
-        [nodesNames]
+        [nodesNames, rootNodeId]
+    );
+
+    const makeSingleReport = useCallback(
+        (reportData) => {
+            if (!Array.isArray(reportData)) {
+                return setNodeName(reportData);
+            } else {
+                if (reportData.length === 1) {
+                    return setNodeName(reportData[0]);
+                } else {
+                    return {
+                        taskKey: GLOBAL_NODE_TASK_KEY,
+                        defaultName: GLOBAL_NODE_TASK_KEY,
+                        reports: [],
+                        subReporters: reportData.map((r) => setNodeName(r)),
+                    };
+                }
+            }
+        },
+        [setNodeName]
     );
```

Comment:
Simplify this function by doing `if (...) { return } something` instead of `if (...) { return } else { something }`